srt-protocol/src/protocol/receiver/mod.rs

```diff
@@ -118,7 +108,7 @@ pub struct Receiver {
     receive_buffer: RecvBuffer,
 
     /// Shutdown flag. This is set so when the buffer is flushed, it returns Async::Ready(None)
-    shutdown_flag: bool,
+    shutdown_flag: Option<Instant>,
```

Comment:
Right, this is a shutdown with a timeout. There were many tests that would fail intermittently/randomly due to injected packet loss or race conditions, this was particularly true of the tokio tests. The connect and rendezvous algorithm tests were failing intermittently for this same reason. We need to implement timeouts for them as well.